chipsec/helper/record/recordhelper.py

```diff
@@ -0,0 +1,335 @@
+            self._data[str(cmd)] = {margs: [str(ret)]}
+
+    def _save(self) -> None:
+        js = dumps(self._data, sort_keys=False, indent=2, separators=(',', ': '))
+        write_file(self._filename, js)
+
+    def _load(self) -> None:
+        if os.path.isfile(self._filename):
+            file_data = read_file(self._filename)
+            if file_data == 0:
+                self._data = {}
+            try:
+                self._data = loads(file_data)
+            except Exception:
+                self._data = {}
+
+    def create(self) -> bool:
+        self.driver_created = True
+        return self._subhelper.create()
+
+    def start(self) -> bool:
+        self._load() # Load file if it exists
+        self.driver_loaded = True
+        return self._subhelper.start()
+
+    def stop(self) -> bool:
+        self._save()
+        self.driver_loaded = False
+        return self._subhelper.stop()
+
+    def delete(self) -> bool:
+        self.driver_created = False
+        return self._subhelper.delete()
+
+    
+
+    def read_pci_reg(self, bus: int, device: int, function: int, offset: int, size: int) -> int:
+        ret = self._subhelper.read_pci_reg(bus, device, function, offset, size)
+        self._add_element("read_pci_reg", (bus, device, function, offset, size), ret)
+        return ret
+
+    def write_pci_reg(self, bus: int, device: int, function: int, offset: int, value: int, size: int) -> int:
+        ret = self._subhelper.write_pci_reg(bus, device, function, offset, value, size)
+        self._add_element("write_pci_reg", (bus, device, function, offset, value, size), ret)
+        return ret
+
+    def read_mmio_reg(self, phys_address: int, size: int) -> int:
+        ret = self._subhelper.read_mmio_reg(phys_address, size)
+        self._add_element("read_mmio_reg", (phys_address, size), ret)
+        return ret
+    
+    def write_mmio_reg(self, phys_address: int, size: int, value: int) -> int:
+        ret = self._subhelper.write_mmio_reg(phys_address, size, value)
+        self._add_element("write_mmio_reg", (phys_address, size, value), ret)
+        return ret
+        
+    #
+    # physical_address is 64 bit integer
+    #
+    def read_phys_mem(self, phys_address:int, length:int) -> bytes:
+        ret = self._subhelper.read_phys_mem(phys_address, length)
+        self._add_element("read_phys_mem", (phys_address, length), ret)
+        return ret
+
+    def write_phys_mem(self, phys_address: int, length: int, buf: bytes) -> int:
+        ret = self._subhelper.write_phys_mem(phys_address, length, buf)
+        self._add_element("write_phys_mem", (phys_address, length, buf), ret)
+        return ret
+
+    def alloc_phys_mem(self, length: int, max_phys_address: int) -> Tuple[int, int]:
+        ret = self._subhelper.alloc_phys_mem(length, max_phys_address)
+        self._add_element("alloc_phys_mem", (length, max_phys_address), ret)
+        return ret
+
+    def free_phys_mem(self, physical_address: int):
```

Comment:
Missing return type

Reply:
In my defense, there is no type hint for this function in the basehelper...

Reply:
Oops, guess we need to add that.  :)